modules/environment/aws/eks/main.tf

```diff
@@ -2,10 +2,14 @@ data "aws_caller_identity" "current" {
 }
 
 locals {
-  ssm_init = <<EOF
+  userdata = <<EOF
 yum install -y amazon-ssm-agent
 systemctl start amazon-ssm-agent
 systemctl enable amazon-ssm-agent
+
+echo '{"registry-mirrors": [${ var.docker_registry_mirror != "" ? format("\"%s\"", var.docker_registry_mirror) : ""}]}' | cat /etc/docker/daemon.json - | jq -s '.[0] * .[1]' > /tmp/daemon.json
```

Comment:
what happens when `registry-mirrors` is set to an empty string?

Reply:
it sets no registry-mirrors `registry-mirrors: []` if we don't handle that case we end up with `registry-mirrors: [ "" ]` which breaks the daemon

Reply:
maybe I'm misreading this, but it seems like `${ var.docker_registry_mirror != "" ? format("\"%s\"", var.docker_registry_mirror) : ""}` will evaluate to `""` if `var.docker_registry_mirror` is set to `""`.

Reply:
yeah, it is hard to read but I did test both cases. if the string is not empty it wraps the string value in quotes "\"http://hostname\"" evaluates to "http://hostname" otherwise "" evaluates to nothing.